src/tracer.ts

```diff
@@ -12,20 +12,31 @@ let tracer: Tracer & { isMock?: boolean } = mockTracer;
  *
  * @param options The `TracerOptions` to be passed the tracer init function
  */
-const init = (options: TracerOptions): void => {
+const init = (
+    options: TracerOptions,
+    plugins: {name: any, config: any}[] = [],
```

Comment:
Does datadog provide a type for the config like they do with tracer options?

Reply:
Internally there is a `Plugins` interface but it is not exported from what I found. I would suggest doing this instead:

```
const init = (
    options: TracerOptions,
    plugins: Record<string, JSONObject>,
): void => {

    tracerOptions = options as DDTracerOptions;

    if (options.useMock !== true) {
        tracer = require('dd-trace');
        tracer.isMock = false;
        if (Object.keys(plugins).length > 0) {
            for (const key in plugins) {
                const value = plugins[key];
                tracer.use(key as any, value);
            }
        }
    }

    tracer.init(tracerOptions);

};
```

this way the name of the plugin is a string and the configuration is expected to be an object, it also prevents redefining the same plugging twice in Eden

add the following under utils.ts

```
export type JSONValue = null | string | number | boolean | { [key: string]: JSONValue } | Array<JSONValue>;
export type JSONObject = { [key: string]: JSONValue };
```